ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts

```diff
@@ -7,30 +8,99 @@
  * This file is part of the OperatorFabric project.
  */
 
-import {Component, Input, OnInit} from '@angular/core';
-import {LineOfLoggingResult} from '@ofModel/line-of-logging-result.model';
-import {TimeService} from '@ofServices/time.service';
-import {Moment} from 'moment-timezone';
+import { Component, Input, OnInit, OnDestroy, OnChanges } from '@angular/core';
+import { LineOfLoggingResult } from '@ofModel/line-of-logging-result.model';
+import { TimeService } from '@ofServices/time.service';
+import { Moment } from 'moment-timezone';
+import { CardService } from '../../../../services/card.service';
+import { selectLoggingCount } from '@ofSelectors/logging.selectors';
+import { Store, select } from '@ngrx/store';
+import { Page } from '@ofModel/page.model';
+import { AppState } from '@ofStore/index';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+import { ExportService } from '@ofServices/export.service';
+
 
 @Component({
     selector: 'of-logging-table',
     templateUrl: './logging-table.component.html',
     styleUrls: ['./logging-table.component.scss']
 })
-export class LoggingTableComponent implements OnInit {
+export class LoggingTableComponent implements OnInit, OnDestroy, OnChanges {
 
 
     @Input() results: LineOfLoggingResult[];
     displayedResult: string;
+    exportLoggingData: Array<any> ;
+    unsubscribe$: Subject<void> = new Subject<void>();
+
+
+    constructor(public timeService: TimeService, private cardService: CardService,
+        private store: Store<AppState>, private translate: TranslateService, private exportService: ExportService) {
+    }
 
-    constructor(public timeService: TimeService) {
+    ngOnChanges() {
```

Comment:
Do not call ExportMonitoringData it for each change and on it , call it just when user want to export (in the export method)